src/System.Runtime.Extensions/src/System/ApplicationId.cs

```diff
@@ -64,30 +65,27 @@ public override string ToString ()
                 sb.Append(ProcessorArchitecture);
                 sb.Append('"');
             }
-            return StringBuilderCache.GetStringAndRelease(sb);
+            return sb.ToString();
         }
-
-        private static char HexDigit(int num) =>
-            (char)((num < 10) ? (num + '0') : (num + ('A' - 10)));
         
         private static string EncodeHexString(byte[] sArray) 
         {
-            string result = null;
-    
-            if (sArray != null)
+            if (sArray == null) return null;
+
+            return string.Create(sArray.Length * 2, sArray, (chars, arr) =>
             {
-                char[] hexOrder = new char[sArray.Length * 2];
-            
-                int digit;
-                for(int i = 0, j = 0; i < sArray.Length; i++) {
-                    digit = (int)((sArray[i] & 0xf0) >> 4);
-                    hexOrder[j++] = HexDigit(digit);
-                    digit = (int)(sArray[i] & 0x0f);
-                    hexOrder[j++] = HexDigit(digit);
+                for (int i = 0, j = 0; i < arr.Length; i++)
+                {
+                    int digit = (arr[i] & 0xf0) >> 4;
+                    chars[j++] = HexDigit(digit);
+
+                    digit = arr[i] & 0x0f;
+                    chars[j++] = HexDigit(digit);
                 }
-                result = new string(hexOrder);
-            }
-            return result;
+            });
```

Comment:
Nice use of string.Create to avoid the char[] allocation.

Reply:
Kudos to @ViktorHofer

Reply:
It would be better to append the characters directly to the ValueStringBuilder, and completely avoid the `char[]` allocation.

Reply:
and done